packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,268 @@
+    if (updateClaimParameters.endDate != 0 && updateClaimParameters.startDate >= updateClaimParameters.endDate)
+      revert IGachaLazyClaim.InvalidDate();
+    if (updateClaimParameters.totalMax < claim.total) revert IGachaLazyClaim.CannotLowerTotalMaxBeyondTotal();
+    if (updateClaimParameters.totalMax > MAX_UINT_32) revert IGachaLazyClaim.InvalidInput();
+    if (updateClaimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    if (updateClaimParameters.cost > MAX_UINT_96) revert IGachaLazyClaim.InvalidInput();
+
+
+    // Overwrite the existing values
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claim.storageProtocol,
+      total: claim.total,
+      totalMax: updateClaimParameters.totalMax,
+      startDate: updateClaimParameters.startDate,
+      endDate: updateClaimParameters.endDate,
+      startingTokenId: claim.startingTokenId,
+      tokenVariations: claim.tokenVariations,
+      location: updateClaimParameters.location,
+      paymentReceiver: updateClaimParameters.paymentReceiver,
+      cost: updateClaimParameters.cost,
+      erc20: claim.erc20
+    });
+    emit GachaClaimUpdated(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaim}.
+   */
+  function getClaim(address creatorContractAddress, uint256 instanceId) public view override returns (Claim memory) {
+    return _getClaim(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaimForToken}.
+   */
+  function getClaimForToken(
+    address creatorContractAddress,
+    uint256 tokenId
+  ) external view override returns (uint256 instanceId, Claim memory claim) {
+    instanceId = _tokenInstances[creatorContractAddress][tokenId];
+    claim = _getClaim(creatorContractAddress, instanceId);
+  }
+
+  function _getClaim(address creatorContractAddress, uint256 instanceId) private view returns (Claim storage claim) {
+    claim = _claims[creatorContractAddress][instanceId];
+    if (claim.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.ClaimNotInitialized();
+  }
+
+  /**
+   * See {IGachaLazyClaim-mintReserve}.
+   */
+  function mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) external payable override {
+    if (Address.isContract(msg.sender)) revert IGachaLazyClaim.CannotMintFromContract();
+    Claim storage claim = _getClaim(creatorContractAddress, instanceId);
+    // Checks for reserving
+    if (mintCount == 0 || mintCount >= MAX_UINT_32) revert IGachaLazyClaim.InvalidMintCount();
+    if (claim.startDate > block.timestamp || (claim.endDate > 0 && claim.endDate < block.timestamp)) revert IGachaLazyClaim.ClaimInactive();
+    if ((claim.totalMax != 0 && claim.total == claim.totalMax)) revert IGachaLazyClaim.ClaimSoldOut();
+    if (claim.total == MAX_UINT_32 || claim.total + mintCount > MAX_UINT_32) revert IGachaLazyClaim.TooManyRequested();
```

Comment:
This can be simplified to:
if (claim.total + mintCount > MAX_UINT_32) revert IGachaLazyClaim.TooManyRequested();

The first part of the if statement is unecessary